client/catalog/src/app/datatypes/DatatypeList.svelte

```diff
@@ -2,67 +2,32 @@
 import client from '../../data/client.js';
 import {Runtime} from '../../data/enums.js';
 import {cap} from '../util.js';
+import HLGrid from '../../ui/HLGrid.svelte';
+import HLGridItem from '../../ui/HLGridItem.svelte';
+import { breadcrumbStore } from '../stores.js';
 
-let load = client.datatype.findMany({	include: {
-		validator: true,
-	}
-});
+let load = client.datatype.findMany({include: {validator: true}});
 
-import { breadcrumbStore } from '../stores.js';
 breadcrumbStore.set(
 	[{
 		href: "/datatypes",
 		text: "Datatypes",
 	}]
 );
-
 </script>
-
-<style>
-	.box {
-		display: flex;
-		flex-direction: row;
-		flex-wrap: wrap;
-	}
-	.stuff {
-
-	}
-	a.object-box {
-		display: flex;
-		flex-direction: column;
-		color: inherit;
-		width: 150px;
-		padding: 1em 1.5em;
-	}
-	a.object-box:hover {
-		background: var(--background-highlight);
-	}
-
-	.spacer {
-		width: 0;
-	}
-	.obj-title{
-		font-weight: 600;
-	}
-
-</style>
-
-<div class="box">
+<HLGrid>
 	{#await load}
 		&nbsp;
 	{:then datatypes}
 		{#each datatypes as datatype}
-			<a href="/datatype/{datatype.id}" class="object-box">
-				<div class="obj-title">{cap(datatype.name)}</div>
+			<HLGridItem type={"datatype"} url={datatype.id} name={datatype.name}>
```

Comment:
splitting the url into type vs url seems a bit arbitrary to me. Why not just take url (or more appropriately, 'href')